stock_replenishment_mrp_bom_selection/wizard/stock_warehouse_orderpoint_replenish_wizard.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2024 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from datetime import datetime
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockWarehouseOrderpointReplenishWizard(models.TransientModel):
+    _name = "stock.warehouse.orderpoint.replenish.wizard"
+    _description = "Stock warehouse orderpoint replenish wizard"
+
+    orderpoint_id = fields.Many2one("stock.warehouse.orderpoint")
+    qty_to_order = fields.Float(related="orderpoint_id.qty_to_order")
+    product_id = fields.Many2one("product.product", related="orderpoint_id.product_id")
+    product_tmpl_id = fields.Many2one(
+        "product.template", related="product_id.product_tmpl_id"
+    )
+    bom_ids = fields.One2many("mrp.bom", related="product_tmpl_id.bom_ids")
+    bom_line_ids = fields.One2many(
+        comodel_name="bom.line.wizard",
+        inverse_name="wizard_id",
+        compute="_compute_bom_line_ids",
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends("orderpoint_id")
+    def _compute_bom_line_ids(self):
+        self.bom_line_ids = None
+        lines_vals = []
+        for line in self.bom_ids:
```

Comment:
You need to filter other variants BoMs:

```suggestion
        boms = self.product_id.bom_ids.filtered(lambda x: not x.product_id or x.product_id == self.product_id)
        for line in boms:
```